test/CMakeLists.txt

```diff
@@ -14,8 +14,18 @@ find_package(benchmark)
 ##############################################################################
 # google test
 
-# remove export symbols from the cxx flags
-string(REPLACE "${EXPORTSYMBOLS}" "" CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
+# remove export symbols from the cxx flags.  This is messy, but there's no `remove_link_flags` function
+get_target_property(os_link_options OS_INTERFACE INTERFACE_LINK_OPTIONS)
+foreach(opt IN ITEMS ${os_link_options})
+  if(opt MATCHES ".*version.script.*.gcc")
+    list(REMOVE_ITEM os_link_options ${opt})
+  endif()
+endforeach()
+unset(opt)
+if(os_link_options)
+  set_target_properties(OS_INTERFACE PROPERTIES INTERFACE_LINK_OPTIONS "${os_link_options}")
+endif()
+unset(os_link_options)
```

Comment:
this way, the -Wl,-export-dynamic part of the EXPORT_SYMBOLS was not being removed. my suggestion would be to re-add the EXPORT_SYMBOLS as a list on the main CMakeLists.txt as: 
```bash 
set(EXPORTSYMBOLS
      -Wl,-export-dynamic
      -Wl,--version-script=${CMAKE_CURRENT_SOURCE_DIR}/exportmap.gcc) 
```  
add that to the OS_INTERFACE:
```bash    
  if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
    # This is only relevant for GCC and causes warnings on Clang
    target_link_options(
      OS_INTERFACE
      INTERFACE
      -Wno-tsan -Wl,-z,relro,-z,now
      ${EXPORTSYMBOLS}
    )
```
and here instead of this foreach loop, use this one:
``` bash 
foreach(symbol ${EXPORTSYMBOLS})
    list(REMOVE_ITEM os_link_options ${symbol})
endforeach()
```